optimum/fx/quantization.py

```diff
@@ -0,0 +1,232 @@
+        # we are assuming people won't configure the model with the type of top level
+        # module here, since people can use "" for global config
+        # We can change this if there is a use case that configures
+        # qconfig using top level module type
+        self.scope = Scope("", None)
+        self.node_name_to_scope: Dict[str, Tuple[str, type]] = {}
+
+    def is_leaf_module(self, m: torch.nn.Module, module_qualified_name: str) -> bool:
+        return (
+            (m.__module__.startswith("torch.nn") and not isinstance(m, torch.nn.Sequential))
+            or module_qualified_name in self.skipped_module_names
+            or type(m) in self.skipped_module_classes
+            or isinstance(m, _FusedModule)
+            or super().is_leaf_module(m, module_qualified_name)
+        )
+
+    def call_module(
+        self, m: torch.nn.Module, forward: Callable[..., Any], args: Tuple[Any, ...], kwargs: Dict[str, Any]
+    ) -> Any:
+        module_qualified_name = self.path_of_module(m)
+        # Creating scope with information of current module
+        # scope will be restored automatically upon exit
+        with ScopeContextManager(self.scope, m, module_qualified_name):
+            return super().call_module(m, forward, args, kwargs)
+
+    def create_node(
+        self,
+        kind: str,
+        target: Target,
+        args: Tuple[Argument, ...],
+        kwargs: Dict[str, Argument],
+        name: Optional[str] = None,
+        type_expr: Optional[Any] = None,
+    ) -> Node:
+        node = super().create_node(kind, target, args, kwargs, name, type_expr)
+        self.node_name_to_scope[node.name] = (self.scope.module_path, self.scope.module_type)
+        return node
+
+    def trace(self, root: "PreTrainedModel", concrete_args: Optional[Dict[str, Any]] = None) -> "Graph":
+        if concrete_args is None and self.specialized_concrete_args is not None:
+            concrete_args = self.specialized_concrete_args
+        return super().trace(root, concrete_args=concrete_args)
+
+
+def specialized_quantization_tracer_creator(concrete_args: Dict[str, Any]) -> Type:
+    """Creates a QuantizationTracer-like class specifying concrete_args as a class attribute."""
+    return type("QuantizationTracer", (QuantizationTracer,), {"specialized_concrete_args": concrete_args})
+
+
+@check_if_available
+def fuse_fx(
+    model: Union["PreTrainedModel", GraphModule],
+    fuse_custom_config_dict: Optional[Dict[str, Any]] = None,
+    input_names: Optional[List[str]] = None,
+    check: bool = True,
+) -> GraphModule:
+    """
+    Transformers models compatible version of torch.quantization.quantize_fx.fuse_fx, refer to PyTorch documentation
+    for more details.
+
+    Args:
+        model (`PreTrainedModel` or `torch.fx.GraphModule`):
+            The model to fuse.
+        fuse_custom_config_dict (`Dict[str, Any]`, *optional*):
+            Refer to PyTorch documentation.
```

Comment:
Could we add a one-liner to explain what this arg is, followed by a link to the PyTorch docs?